File: yujin-level0/Week68/캐릭터의 좌표.js
function solution(keyinput, board) {
  // 현재 좌표 초기화 (루프 바깥에 선언)
  let x = 0;
  let y = 0;

  // 보드 크기에 따른 최대 이동 가능 범위 계산 (절댓값 기준)
  const maxX = Math.floor(board[0] / 2);
  const maxY = Math.floor(board[1] / 2);

  // 방향키 입력 순회
  for (let dir of keyinput) {
    if (dir === "left" && x > -maxX) {
      x -= 1;
    } else if (dir === "right" && x < maxX) {
      x += 1;
    } else if (dir === "up" && y < maxY) {
      y += 1;
    } else if (dir === "down" && y > -maxY) {
      y -= 1;
    }
  }

  // 최종 좌표 [x, y] 반환
  return [x, y];
}

// 다른 풀이
function solution(keyinput, board) {
  let res = [0, 0];

  const maxX = Math.floor(board[0] / 2);
  const maxY = Math.floor(board[1] / 2);
  for (let p of keyinput) {
    switch (p) {
      case "left":
        if (-res[0] < maxX) res[0]--;
        break;
      case "right":
        if (res[0] < maxX) res[0]++;
        break;
      case "up":
        if (res[1] < maxY) res[1]++;
        break;
      case "down":
        if (-res[1] < maxY) res[1]--;
        break;
    }
  }
  return res;
}

File: gnsals0904/level0/캐릭터의_좌표.java
class Solution {
    static int garo, sero;
    public int[] solution(String[] keyinput, int[] board) {
        garo = board[0]/2;
        sero = board[1]/2;
        Point p = new Point(0, 0);
        for(String key : keyinput){
            p.move(key);
        }
        int[] result = new int[2];
        result[0] = p.x;
        result[1] = p.y;
        return result;
    }
    
    class Point{
        int x;
        int y;
        
        Point(int x, int y){
            this.x = x;
            this.y = y;
        }
        
        void move(String dir){
            if(dir.equals("left")){
                x--;
            } else if(dir.equals("right")){
                x++;
            } else if(dir.equals("up")){
                y++;
            } else if(dir.equals("down")){
                y--;
            }
            if(x < garo*-1) x = garo*-1;
            if(x > garo) x = garo;
            if(y < sero*-1) y = sero*-1;
            if(y > sero) y = sero;
        }
    }
    
    
}
